inspirehep/modules/records/validators/helpers.py

```diff
@@ -0,0 +1,88 @@
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Helpers for validators module."""
+
+from __future__ import absolute_import, division, print_function
+
+from invenio_db import db
+from jsonschema import ValidationError
+from collections import deque
+
+FIELD_VALUE_REQUIRE_FIELD_TEMPLATE = "'{field}' field value '{value}' requires field '{required}' to exist."
+FIELD_REQUIRE_FIELD_VALUES_TEMPLATE = "'{field}' field requires '{required}' field to have " \
+                                      "at least one of the values {values}."
+FIELD_REQUIRE_FIELD_TEMPLATE = "'{field}' field requires '{required}' field to exist."
+FIELD_DUPLICATE_VALUES_FOUND_TEMPLATE = "'{field}' field with value '{value}' found in more than one record."
+GET_RESULTS_FOR_FIELD_PROPERTY_QUERYSTRING_TEMPLATE = "SELECT id FROM records_metadata as r, " \
+                                                      "json_array_elements(r.json -> '{field}') " \
+                                                      "as elem WHERE elem ->> '{prop}'='{value}'"
+FIELD_VALIDATION_TEMPLATE = "Field '{field}' with value '{value}' is not valid."
+SCHEMA_FIELDS_THAT_CAN_CONTAIN_DATE = {
+    '_desy_bookkeeping': ['date'],
+    'imprints': ['date'],
+    'thesis_info': ['date', 'defense_date'],
+    '_fft': ['creation_datetime'],
+    'legacy_creation_date': [],
+    'preprint_date': []
+}
+
+
+def check_field_values_not_in_required_values_for_record(record, field, required_values):
+    field_values = record.get(field, [])
+    common_list = [x for x in field_values if x in required_values]
+    return not common_list
```

Comment:
This is an intersection of sets, so it can be written as one. See: https://github.com/inspirehep/inspire-next/blob/a718afaac615b555ad5f524e781a731cddfc015d/inspirehep/modules/workflows/tasks/matching.py#L231-L235